src/mlpack/tests/loss_functions_test.cpp

```diff
@@ -711,4 +712,63 @@ BOOST_AUTO_TEST_CASE(MarginRankingLossTest)
       "-0.753830 1.336900 0.000000 0.000000 -0.207000 0.328810"), 1e-6);
 }
 
+/**
+ * Test for the multiLabel softmargin loss function, where loss is
+ * reduced to sum.
+ */
+BOOST_AUTO_TEST_CASE(MultiLabelSoftMarginLossSumReductionTest)
+{
+  arma::mat input, target, output;
+  double loss;
+  arma::mat weight;
+  weight.ones(1, 3);
+  MultiLabelSoftMarginLoss<> module(weight, 3, true);
+
+  // Test the Forward function. Loss should be 2.14829.
+  input << 0.1778 << 0.1203 << -0.2264 << arma::endr
+      << 0.0957 << 0.2403 << -0.3400 << arma::endr
+      << 0.1397 << 0.1925 << -0.3336 << arma::endr;
+  target << 0 << 1 << 0 << arma::endr
+      << 1 << 0 << 0 << arma::endr
+      << 0 << 0 << 1 << arma::endr;
```

Comment:
This seems like a lot of lines. Could you refer to how matices are declared in other tests. I think that looks cleaner. Let me know what you think.

Reply:
Changed it to match the format for other test cases :)

Reply:
I am probably missing something somewhere. But, isn't this the primary way to initialize a 2D matrix with particular values in each cell?

Reply:
You could do the following:

```
 mat = arma::mat("1.0000 1.000 1.000 2.0000 2.0000 \
      2.0000 2.000 2.5000 3.0000 3.0000");
 mat.reshape(5, 2);
```

I think I haven't seen this declaration in any file other than convolution_test (not the layer)